src/components/utils/include/utils/timer.h

```diff
@@ -44,20 +44,61 @@ namespace timer {
 
 typedef uint32_t Milliseconds;
 
+/**
+ * @brief Timer calls custom callback function after
+ * specified timeout has been elapsed
+ */
 class Timer {
  public:
+  /**
+   * @brief Constructor
+   * Does not starts timer
+   * @param name Timer's name for identity
+   * @param task Task for tracking
+   */
   Timer(const std::string& name, TimerTask* task);
+
+  /**
+   * @brief Destructor
+   * Stops timer if it's running
+   */
   ~Timer();
   
+  /**
+   * @brief Starts timer with specified timeout
+   * @param timeout Timer timeout
+   * @param single_shot Shows needs to restart timer after timeout
+   */
   void Start(const Milliseconds timeout, const bool single_shot);
+
+  /**
+   * @brief Stops timer if it's running
+   */
   void Stop();
 
+  /**
+   * @brief Gets current timer status
+   * @return True in case of timer is running, false otherwise
+   */
   bool is_running() const;
+
+  /**
+   * @brief Gets current timer timeout
+   * @return Current timeout in milliseconds.
+   * Null if timer has not been started
+   */
   Milliseconds timeout() const;
 
  private:
+  /**
+   * @brief Delegate for timer thread
+   */
   class TimerDelegate : public threads::ThreadDelegate {
    public:
+    /**
+     * @brief Constructor
+     * @param timer Timer instance pointer for callback calling
+     */
     explicit TimerDelegate(const Timer* timer);
 
     void set_timeout(const Milliseconds timeout);
```

Comment:
@anosach-luxoft, 
Maybe also provide doxygen?
For example, Is this time includes `OntImeout` logics routine?
What will be in case of 0 or maximum uint32 value? 
Or seuggetion to follow Timer tinout value